develop/tutorials/articles/200-integrating-with-front-end-frameworks/01-integrating-with-react.markdown

```diff
@@ -0,0 +1,201 @@
+# Integrating with React [](id=integrating-with-react)
+
+You can use the 
+[npm React portlet template](/develop/reference/-/knowledge_base/7-1/npm-react-portlet-template) 
+to automate much of the required configuration for you, or create the module 
+manually if you wish. For convenience, all steps are listed if you want to 
+proceed with a manual configuration. This tutorial shows how to use React in 
+your portlets, whether you're migrating an existing React project or building a 
+fresh project. See the 
+[npm React portlet template](/develop/reference/-/knowledge_base/7-1/npm-react-portlet-template) 
+reference docs for more information on the portlet's anatomy or the 
+[react npm portlet sample](/develop/reference/-/knowledge_base/7-1/react-npm-portlet) 
+for a React portlet example that you can test and deploy right now. Get started 
+by creating your OSGi module and configuring its metadata. 
+
+## Configuring Metadata [](id=configuring-metadata)
+
+Follow these steps to create the module and configure its metadata for React:
+
+1.  [Create an OSGi module](/develop/tutorials/-/knowledge_base/7-1/starting-module-development#creating-a-module). 
+    For example, use the 
+    [npm React portlet template](/develop/reference/-/knowledge_base/7-1/npm-react-portlet-template). 
+
+2.  Specify the `Web-ContextPath` BND Header in your project's `bnd.bnd` file. 
+    Below is the default configuration for the npm React portlet template:
+
+        Web-ContextPath: /my-npm-react-portlet
+
+3.  Create a 
+    [`.babelrc` file](/develop/reference/-/knowledge_base/7-1/configuring-liferay-npm-bundler) 
+    and add the following presets to it:
+
+        {
+        	"presets": ["es2015", "react", "liferay-project"]
+        }
+
+4.  Add a 
+    [`.npmbundlerrc` file](/develop/reference/-/knowledge_base/7-1/configuring-liferay-npm-bundler) 
+    to your project's root folder and add the 
+    [`liferay-npm-bundler-preset-react` preset](https://www.npmjs.com/package/liferay-npm-bundler-preset-react). 
+    This preset provides the required `.npmbundlerrc` configuration. You can, 
+    however, copy the contents of the preset and add additional configuration 
+    options to the file instead if needed:
+
+        {
+            "preset": "liferay-npm-bundler-preset-react"
+        }
+
+5.  Include the following dependency to your `build.gradle` file:
+
+        compileOnly group: "com.liferay", 
+        name: "com.liferay.frontend.js.loader.modules.extender.api", 
+        version: "2.0.2"
+
+6.  Create a `package.json` in your project if it doesn't already exist and add 
+    the configuration shown below to it. Update the `"main"` JS path to reflect 
+    your app's main JS file (remember to add the `.es.js` extension if your file 
```

Comment:
I wouldn't say anything about `.es.js` extension, as it is not standard and doesn't serve any purpose.